sycl/unittests/pi/PlatformTest.cpp

```diff
@@ -6,80 +6,94 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <CL/sycl.hpp>
 #include <CL/sycl/detail/pi.hpp>
 #include <gtest/gtest.h>
-#include <memory>
+#include <vector>
+
+namespace {
 
 using namespace cl::sycl;
 
-namespace pi {
 class PlatformTest : public ::testing::Test {
 protected:
+  pi_uint32 platform_count;
 
-constexpr static size_t out_string_size =
-    8192u; // Using values from OpenCL CTS clGetPlatforms test
+  PlatformTest() : platform_count{0u} { detail::pi::initialize(); };
 
-  PlatformTest() { detail::pi::initialize(); }
+  ~PlatformTest() override = default;
 
-  ~PlatformTest() = default;
-};
+  void SetUp() {
+    ASSERT_NO_FATAL_FAILURE(Test::SetUp());
 
-TEST_F(PlatformTest, piPlatformsGet) {
-  pi_uint32 platformCount = 0;
+    const static char *platform_count_key = "PiPlatformCount";
 
-  ASSERT_EQ(PI_CALL_RESULT(RT::piPlatformsGet(0, 0, &platformCount)),
-            PI_SUCCESS)
-      << "piPlatformsGet failed";
+    // Initialize the logged number of platforms before the following assertion.
+    RecordProperty(platform_count_key, platform_count);
 
-  ASSERT_GT(platformCount, 0u) << "piPlatformsGet found 0 platforms.\n";
+    ASSERT_EQ(RT::piPlatformsGet(0, 0, &platform_count), PI_SUCCESS);
 
-  std::vector<pi_platform> platforms(platformCount);
+    // Overwrite previous log value with queried number of platforms.
+    RecordProperty(platform_count_key, platform_count);
 
-  ASSERT_EQ(PI_CALL_RESULT(
-                RT::piPlatformsGet(platformCount, platforms.data(), nullptr)),
+    if (platform_count == 0u) {
+      std::cout << "WARNING: Unable to find platforms with piPlatformsGet.\n";
```

Comment:
Can we somehow stop any furhter the testing here and not have the rest of the test(s) specially handle the platform_count? Maybe, we could instead create/keep std::vector<pi_platform>, and have the rest of the tests iterate over that, which will just do nothing for an empty vector.

Reply:
I will look into moving even more code into the `SetUp` member function of the fixture.

I am experimenting with making other unittests use gtest's value parameterized testing approach which won't run tests if there are no platforms/devices.

This platform testing does not use that approach to make the platform usage more explicit while it is more of a fixture/scaffolding for other tests.

Reply:
Done.

Reply:
I am not aware of a way to stop further testing in gtest other than failing the test (the latest gtest versions 1.1.x have a way to skip tests but that is not in the version that LLVM comes with (1.8.0)).

Instead what now happens is that one test is the `SetUp` and the other test loops over the found programs - so does not do anything on platforms without PI platforms.